relay-general/src/store/normalize.rs

```diff
@@ -2130,6 +2136,44 @@ mod tests {
         "###);
     }
 
+    #[test]
+    fn test_filter_invalid_measurement_name() {
+        let json = r#"
+        {
+            "type": "transaction",
+            "timestamp": "2021-04-26T08:00:05+0100",
+            "start_timestamp": "2021-04-26T08:00:00+0100",
+            "measurements": {
+                "my_custom-measurement_1": {"value": 123},
+                "my_custom_measurement_2": {"value": 789}
```

Comment:
nit: considering the parsing fails if the name is not correct and we've modified the accepted domain, I'd add a third measurement with a `.` (valid), and maybe one with a `/` (invalid).

Reply:
This was great advice! By adding some more cases I realized that there is already protocol validation in `protocol.rs`, which actually adds `errors` for invalid names, so I reverted all changes in this PR and just fixed the validation there.